lib/ansible/cli/vault.py

```diff
@@ -133,7 +133,12 @@ def run(self):
             self.b_new_vault_pass = CLI.read_vault_password_file(self.options.new_vault_password_file, loader)
 
         if not self.b_vault_pass or self.options.ask_vault_pass:
-            self.b_vault_pass = self.ask_vault_passwords()
+            # the 'read' options dont need to ask for password confirmation.
+            # 'edit' is read/write, but the decrypt will confirm.
+            if self.action in ['decrypt', 'edit', 'view', 'rekey']:
```

Comment:
'rekey' confirms the 'new' password later. The prompt for the existing old password does not need to confirm.

There is no 'new'.  Do you mean 'create' or 'view'?